### ISSUE TITLE
Pricing ambiguity for same model names from different providers in TokenCounter

### ISSUE THREAD

**Issue by lyie28** [lavague-ai/LaVague#496]
I think as we add more models we will have some issues between how we know the pricing of a model name that is provided by two different providers with different pricing. For example, we should use pricing from OpenAI or Azure etc., but currently we just take the model name 'gpt4-o'

**Comment by paulpalmieri** [comment]
Agreed, here are some other improvements that could lead to better token cost estimations
- Use more precise tokenizers instead of a [default one for everything ](https://github.com/lavague-ai/LaVague/blob/5147bfd3504eaa6808afa42c39f738440f53fa39/lavague-core/lavague/core/token_counter.py#L29)
    - This can currently lead to a small margin of error (10/15%) in our token count estimates. 
    - Reason for this is that there's no general purpose tokenizer libraries and `TokenCountingHandler` requires tokenizers to be passed. 
    - We'd have to integrate each individual tokenizer provider
- As we add models and keep using the trick above, we should:
    - Try to find tokenizers for Claude, Llama, etc. and integrate them
    - Or at least try to compare the output of our default tokenizer with an exact tokenizer to get an idea of the mulitplier we could apply on our estimates. 
- Make sure that images are counted properly by our implementation relying on llamaindex callbacks
